example-app/SantokuApp/src/bases/ui/common/StyledFlatList.tsx

```diff
@@ -0,0 +1,23 @@
+import {BoxProps, createBox} from '@shopify/restyle';
+import React from 'react';
+import {FlatList, FlatListProps} from 'react-native';
+
+import {RestyleTheme} from '../theme/restyleTheme';
+
+// createBoxでは、FLatListPropsのジェネリクスの型定義ができない
+// https://github.com/Shopify/restyle/issues/166
+export const BaseFlatList = createBox<RestyleTheme, FlatListProps<any>>(FlatList);
+
+type StyledFlatListProps<T> = BoxProps<RestyleTheme> & FlatListProps<T>;
+const StyledFlatListComponent = <T extends unknown>(
+  {children, ...rest}: StyledFlatListProps<T>,
+  ref: React.ForwardedRef<FlatList>,
+) => (
```

Comment:
How about putting together a definition.

```suggestion
type StyledFlatListProps<T> = BoxProps<RestyleTheme> & FlatListProps<T> & {ref?: React.ForwardedRef<FlatList>};
const StyledFlatListComponent = ({children, ref, ...rest}: StyledFlatListProps<unknown>) => (
```